packages/runtime-common/schema-analysis-plugin.ts

```diff
@@ -545,3 +545,27 @@ function insertOrReplace(item: Declaration, arr: Declaration[]) {
   }
   return arr;
 }
+
+export function isEquivalentBody(
+  path: NodePath | undefined,
+  newPath: NodePath | undefined,
+) {
+  if (
+    path?.node &&
+    newPath?.node &&
+    'body' in path.node &&
+    'body' in newPath.node &&
+    path.node.body &&
+    newPath.node.body &&
+    'loc' in path.node.body &&
+    'loc' in newPath.node.body &&
+    path.node.body.loc &&
+    newPath.node.body.loc
+  ) {
+    let { start: newStart, end: newEnd } = newPath.node.body.loc;
+    let { start, end } = path.node.body.loc;
+    // @ts-expect-error Property 'token' does not exist on type '{ line: number; column: number; }'.
+    return start.token === newStart.token && end.token === newEnd.token;
```

Comment:
Is it really correct to say that two bodies of code are the same when their first and last tokens are equal?

Reply:
I mean, would it be more appropriate to say "has same position" instead of "has equivalent body"?

Reply:
No it does not. 

It does not take into account the content of the body rather the position of the body in code. So perhaps not equvialentBody but equvialentBodyPosition. 

Here is an example 

```
function greet(name) {
  console.log(`Hello, ${name}!`);
}
```

body
```
{
  console.log(`Hello, ${name}!`);
}
```

tokens 
```
start token = { 
end token = }
```

new code 

```
function summon(name) {
  console.log(`Hello, ${name}!`);
}
```

body
```
{
  console.log(`Hello, ${name}!`);
}
```

tokens 
```
start token = { 
end token = }
```

Reply:
> I mean, would it be more appropriate to say "has same position" instead of "has equivalent body"?

yea this makes sense. Ill change it